api/server/router/container/backend.go

```diff
@@ -43,7 +43,7 @@ type stateBackend interface {
 	ContainerStop(name string, seconds int) error
 	ContainerUnpause(name string) error
 	ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error)
-	ContainerWait(name string, timeout time.Duration) (int, error)
+	ContainerWait(name string, timeout time.Duration, waitForRemoval bool) (int, error)
```

Comment:
Yes, that's what I meant. And this is the current behavior today.

I agree we need to change it, but the proper way to do it would be to wait for your detach PR to get merged so we can wait on that event instead of having an arbitrary timeout :).